app/src/main/java/com/nextcloud/talk/jobs/CapabilitiesWorker.java

```diff
@@ -67,44 +71,45 @@ public class CapabilitiesWorker extends Worker {
 
     NcApi ncApi;
 
+    private Disposable dbQueryDisposable;
+
     public CapabilitiesWorker(@NonNull Context context, @NonNull WorkerParameters workerParams) {
         super(context, workerParams);
 
     }
 
-    private void updateUser(CapabilitiesOverall capabilitiesOverall, UserEntity internalUserEntity) {
+    private void updateUser(CapabilitiesOverall capabilitiesOverall, User internalUserEntity) {
         try {
-            userUtils.createOrUpdateUser(null, null,
-                    null, null,
-                    null, null, null, internalUserEntity.getId(),
-                    LoganSquare.serialize(capabilitiesOverall.getOcs().getData().getCapabilities()), null, null)
-                    .blockingSubscribe(new Observer<UserEntity>() {
-                        @Override
-                        public void onSubscribe(Disposable d) {
-
-                        }
-
-                        @Override
-                        public void onNext(UserEntity userEntity) {
-                            eventBus.post(new EventStatus(userEntity.getId(),
-                                    EventStatus.EventType.CAPABILITIES_FETCH, true));
-                        }
-
-                        @Override
-                        public void onError(Throwable e) {
-                            eventBus.post(new EventStatus(internalUserEntity.getId(),
-                                    EventStatus.EventType.CAPABILITIES_FETCH, false));
-                        }
-
-                        @Override
-                        public void onComplete() {
-
-                        }
-                    });
+            dbQueryDisposable = userManager.createOrUpdateUser(null, new UserManager.UserAttributes(
+                internalUserEntity.getId(),
+                null,
+                null,
+                null,
+                null,
+                null,
+                null,
+                LoganSquare.serialize(capabilitiesOverall.getOcs().getData().getCapabilities()),
+                null,
+                null)).subscribe(
+                user -> eventBus.post(new EventStatus(user.getId(),
+                                                      EventStatus.EventType.CAPABILITIES_FETCH,
+                                                      true)),
+                throwable -> {
+                    eventBus.post(new EventStatus(internalUserEntity.getId(),
+                                                  EventStatus.EventType.CAPABILITIES_FETCH,
+                                                  false));
+                    dispose(dbQueryDisposable);
+                },
+                () -> dispose(dbQueryDisposable));
```

Comment:
@AlvaroBrey any idea/hint what I need to do to make this a synchronous call, currently everything throughout the app is `rxjava2` which doesn't offer a `blockingSubscribe` :(

Reply:
Any reason why you want to make it synchronous at all? I don't see any need

Reply:
While not having looked into manually testing it, I would think that it won't work async